errors.go

```diff
@@ -45,29 +59,30 @@ type ExitCoder interface {
 	ExitCode() int
 }
 
-// ExitError fulfills both the builtin `error` interface and `ExitCoder`
-type ExitError struct {
+type exitError struct {
 	exitCode int
 	message  interface{}
 }
 
-// NewExitError makes a new *ExitError
-func NewExitError(message interface{}, exitCode int) *ExitError {
-	return &ExitError{
-		exitCode: exitCode,
+// NewExitError makes a new *exitError
+func NewExitError(message interface{}, exitCode int) ExitCoder {
+	return Exit(message, exitCode)
+}
+
+// Exit wraps a message and exit code into an ExitCoder suitable for handling by
+// HandleExitCoder
+func Exit(message interface{}, exitCode int) ExitCoder {
```

Comment:
❓ _(blocking question)_ `NewExitError` and `Exit` are exactly the same function, yeah? I can't tell why `Exit` is being added in this PR?

Reply:
Good question. This is how it was named in v2 and I have followed v2 names everywhere. At places like this, I'd be glad to revert to make it more compatible with v1 and one less point for migration.

Good catch (Y)

Reply:
It's frustrating because the new name (`Exit`) is definitely better, but we can easily make changes like this in v `2.1` or `2.9` or whatever. So it'll make our lives easier to remove these kinds of changes from this V2 PR, yeah 👍

Reply:
If we can make this in 2.1 then I wonder why not in 2.0? I think we should have a new 2.0 release with all the features

Reply:
That's fine 👍 but I would like the record to state that I predict users will say _"V2 is changing too much"_